worker-core/src/main/java/com/hpe/caf/worker/core/StreamingWorkerWrapper.java

```diff
@@ -48,8 +48,11 @@ public void run()
     {
         try {
             if (workerTask.isPoison()) {
-                LOG.warn("Worker [" + worker.getWorkerIdentifier() + "] did not handle poisoned message, when it was passed for processing.");
-                throw new RuntimeException("Worker [" + worker.getWorkerIdentifier() + "] did not handle poisoned message, when it was passed for processing.");
+                final String message = "Worker [" + worker.getWorkerIdentifier()
+                + "] did not handle poisoned message, when it was passed for processing. "
+                + workerTask.getTaskMessage();
+                LOG.warn(message);
```

Comment:
The message contains the taskData which could contain sensitive information from the user.
The taskData could also be quite big.

Reply:
I removed taskData from the log but included it in the exception message. Without the taskData not sure if the message would have sufficient details to understand what caused the failure.

Reply:
Great, it was the logging line above the comment was related to.
![image](https://user-images.githubusercontent.com/18459684/88376191-d32cbd80-cd94-11ea-947c-da8ea968e0ca.png)